edi_sale_oca/components/process.py

```diff
@@ -0,0 +1,86 @@
+# Copyright 2021 Camptocamp SA
+# @author: Simone Orsi <simone.orsi@camptocamp.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+
+from odoo import _
+from odoo.exceptions import UserError
+
+from odoo.addons.component.core import Component
+
+
+class EDIExchangeSOInput(Component):
+    """Process sale orders."""
+
+    _name = "edi.input.sale.order.process"
+    _inherit = "edi.component.input.mixin"
+    _usage = "input.process.sale.order"
+
+    def __init__(self, work_context):
+        super().__init__(work_context)
+        self.settings = {}
+        process_ctx = self.type_settings.get("process", {})
+        self.settings = process_ctx.get("env_ctx", {})
```

Comment:
You don't need this. The `env_ctx` is already propagated to the collection's context and to the record context.
The one of the collection is the one you have on self.env since the env comes from the collection.
See below.